### PR TITLE
Fluid Supplier Overhaul

### PR DESCRIPTION
**Pull request by TimeConqueror**
Old GUI:
image

New GUI with ModularUI2:
image

- Switched to ModularUI2 to enhance user experience with hotkeys, click-features, etc.
- Replaced Fluid amount from buttons to textfield
- Replaced Min.Mode from button to textfield (refill_on_depletion)
- Added tooltips, which explain how pipe works  
image
image

Other changes: optimized Fluid Supplier algorythms

### CODE REVIEW

File: src/main/java/logisticspipes/compat/ModularUIHelper.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            .location(LogisticsPipes.rl("textures/gui/GuiBackground.png")).imageSize(45, 45).adaptable(15).build();
+
+    public static void openPipeUI(EntityPlayer player, CoreUnroutedPipe pipe) {
+        World world = pipe.getWorld();
+        if (world == null || world.isRemote) return;
+
+        TileEntityGuiFactory.open(player, pipe.getX(), pipe.getY(), pipe.getZ());
+    }
+
+    public static IKey reallyDynamicKey(Supplier<IKey> keySupplier) {
```
Comment (miozune) [inline src/main/java/logisticspipes/compat/ModularUIHelper.java:27]: You can use `DynamicDrawable` since `overlay` accepts `IDrawable`, not just `IKey`

Comment (miozune) [reply src/main/java/logisticspipes/compat/ModularUIHelper.java:27]: Oh right, that's why you requested dynamic `LangKey`. Now you can use it 😉

**Review by Glease — APPROVED** [review]
skimmed through the code and there is no obvious problems. if it works in game then it'd be good to merge

File: src/main/java/logisticspipes/compat/ModularUIHelper.java (hunk 1 above)
Comment (miozune) [reply src/main/java/logisticspipes/compat/ModularUIHelper.java:27]: Oh I meant "it" by new supplier for `LangKey`, not `reallyDynamicKey`... English is hard...

**Pull request merged**